code/modules/mob/living/living_health_procs.dm

```diff
@@ -383,11 +383,8 @@
 
 ///Revive the huamn up to X health points
 /mob/living/carbon/human/proc/revive_to_crit(should_offer_to_ghost = FALSE, should_zombify = FALSE)
-	if(!has_working_organs())
-		on_fire = TRUE
-		fire_stacks = 15
-		update_fire()
-		QDEL_IN(src, 1 MINUTES)
+	if(on_fire || !has_working_organs())
+		species.handle_death(src)
```

Comment:
why is this calling species handle death? why isnt it calling normal death or dealing damage instead?